src/server.tsx

```diff
@@ -103,7 +104,11 @@ const buildEpic = async (
     );
 
     logTargeting(`Renders Epic true for targeting: ${JSON.stringify(targeting)}`);
-    return { html, css };
+
+    // TODO: determine if we need init code in a more elegant fashion
+    const needsInit = !!variant.showReminderFields;
```

Comment:
Another approach would be to let the ContributionsEpic component decide and return what JS is needed. To do this, we'd have to turn it into a function returning a component and (optional) JS.

This would avoid the implicit coupling between the markup ContributionsEpic returns, and the JS we decide on here, which seems a bit error prone and is harder to test together. (For eaxmple, to avoid serving JS that doesn't relate to the markup.)

I think the assumption at the moment (`needsInit`) is that there will only ever be a single JS included (or not). But I suspect that isn't true.

Reply:
Yeah I agree it's the responsibility of the component to decide which JS should run to initialise the component. I've made some changes to this so that:
- The default export of the component is now an object with the component name (we can add other meta data in the future), the component itself and an optional `getInitScript` function that can be called to return a JS string to be sent down and evaluated on the client.

Then in `server.tsx` when we extract the HTML & CSS, we also try to extract the JS if the `getInitScript` function is defined in the component. So going forward any component can have the 3 properties (html/css/js).

Also updated the `GET /epic` route handler so that we can run this JS on the browser when viewing the component. Sadly, don't think we'll be able to do the same in Storybook but I'll keep looking for a workaround. @nicl